src/settings/s_options.php

```diff
@@ -91,6 +91,15 @@ static function render_display_property(SettingValues $sv, PaperOption $o, $xpos
             . $sv->render_messages_at("optdt_$xpos")
             . "</div>";
     }
+    static function render_required_property(SettingValues $sv, PaperOption $o, $xpos, $self, $gj) {
+        $self->add_option_class("fold8" . ($o->required ? "o" : "c"));
+        return '<div class="' . $sv->control_class("optreq_$xpos", "entryi short fx8")
+            . '">' . $sv->label("optreq_$xpos", "Required")
+            . '<span class="sep">'
+            . Ht::select("optreq_$xpos", ["true" => "True", "" => "False"], $o->required ? "true" : "", ["class" => "settings-opt-required", "id" => "optreq_$xpos"])
+            . $sv->render_messages_at("optreq_$xpos")
+            . "</span></div>";
+    }
```

Comment:
The HotCRP.com version is as follows. Rather than add another icon, it uses the existing “form status” icon (the pencil).

```
    static function render_required_property(SettingValues $sv, PaperOption $o, $xpos, $self, $gj) {
        $self->add_option_class("fold9" . ($o->required ? "o" : "c"));
        return '<div class="' . $sv->control_class("optreq_$xpos", "entryi short fx9")
            . '">' . $sv->label("optreq_$xpos", "Required")
            . Ht::select("optreq_$xpos", ["" => "No", "1" => "Yes"], $o->required ? "1" : "", ["id" => "optreq_$xpos"])
            . $sv->render_messages_at("optreq_$xpos")
            . "</div>";
    }
```

Any comments on that choice?

Reply:
That keeps the UI simpler, and I agree that the two fields are related. Fine by me